app/models/hackney/task.rb

```diff
@@ -0,0 +1,73 @@
+class Hackney::Task
+  include ActiveModel::Model
+
+  # FIXME: Task and WorkOrder are somewhat mixed in the API
+
+  attr_accessor :sor_code
+  attr_accessor :sor_code_description
+  attr_accessor :trade
+  attr_accessor :work_order_reference
+  attr_accessor :repair_request_reference
+  attr_accessor :problem_description
+  attr_accessor :created_at
+  attr_accessor :auth_date
+  attr_accessor :estimated_cost
+  attr_accessor :actual_cost
+  attr_accessor :completed_at
+  attr_accessor :date_due
+  attr_accessor :work_order_status
+  attr_accessor :dlo_status
+  attr_accessor :servitor_reference
+  attr_accessor :property_reference
+  attr_accessor :supplier_reference
+  attr_accessor :user_login
+  attr_accessor :user_name
+  attr_accessor :authorised_by
+  attr_accessor :estimated_units
+
+  def self.for_work_order(work_order_reference)
+    HackneyAPI::RepairsClient
+      .new
+      .get_work_order_tasks(work_order_reference)
+      .map do |api_attributes|
+      new_from_api(api_attributes)
```

Comment:
should this be inlined?

Reply:
No, `new_from_api` is a public method and should be available to other models design. For example, if another API response embed a `Task`, then can use it to parse just that bit.

Reply:
Oh no, sorry I meant one more tab in

Reply:
These method call needs to be properly indented

How about:

```
def self.for_work_order(work_order_reference)
  HackneyAPI::RepairsClient.new
    .get_work_order_tasks(work_order_reference).map do |api_attributes|
      new_from_api(api_attributes)
    end
end
```

Reply:
hmm yess this is not very pretty indeed.

Reply:
I'd rather have it liek this:

```
    HackneyAPI::RepairsClient
      .new
      .get_work_order_tasks(work_order_reference)
      .map { |api_attributes| new_from_api(api_attributes) }
```

should we make a poll?

Reply:
I've pushed my proposal :tongue:.